fdbclient/SystemData.cpp

```diff
@@ -328,6 +329,18 @@ UID serverKeysDecodeServer(const KeyRef& key) {
 	rd >> server_id;
 	return server_id;
 }
+
+std::pair<UID, Key> serverKeysDecodeServerBegin(const KeyRef& key) {
```

Comment:
This comment applies here: https://github.com/apple/foundationdb/pull/6686/files/992ff59bfbf84374e8b76ad29b12f93660e7e3e8#r866153546

Reply:
> I think this would be cleaner if a BinaryReader::size() method were provided. Then this could simply be something like:
>>auto const bytesToRead = rd.size();
return { server_id, Key(rd.arenaRead(bytesToRead), bytesToRead) };

I think the constructor of `Key` only accepts the `KeyRef`as parameter, so first getting a KeyRef is necessary.